src/django_components/component.py

```diff
@@ -1306,6 +1306,8 @@ def _render_impl(
         # The component rendering was short-circuited by an extension, skipping
         # the rest of the rendering process. This may be for example a cached content.
         if result_override is not None:
+            # Cleanup needs to be done even if short-circuited
+            context.render_context.pop()
```

Comment:
@oliverhaas Looks good to me, nice find!

I wonder if we could move the calling of `context.render_context.push()` from [line 1269](https://github.com/django-components/django-components/pull/1135/files#diff-35c84ebceae50f7149f03bbcaf2b4f1289bc2984bf7859ca521b5dbe7960c396R1269) to below this point. That way we wouldn't have to call `context.render_context.pop()`.

But also the compatibility with Django's `{% block %}` tags is a bit of a black magic, so tbh I'd leave things as they are, and use this fix.